### PR TITLE
[BUILD] cmake: search external dependencies part moved to separate file

### PR DESCRIPTION
**Pull request by KrupsonSVK**
skuste ci to frci, je umiestnenie noveho cmaku filu ok?

### CODE REVIEW

**Review by knapikm — APPROVED** [review]
po oprave funguje

**Review by kapecp — CHANGES_REQUESTED** [review]
doxygen, sphinx, astyle, cpplint, cppcheck - presunut do samostatnych .cmake suborov

- su to ine veci ako find* skripty, pretoze cez add_custom_target pridavaju nove build target-y

**Comment by KrupsonSVK** [comment]
nerozumiem, nebolo to vo find* skripte. ak dobre chapem s oddelenim tych add_target do samostatnych cmake, tak napr takto:

_searchDependencies.cmake:_

> option(SKIP_ASTYLE "Skip using Astyle code formatter" OFF)
if (NOT SKIP_ASTYLE)
    find_package(Astyle)
    if (Astyle_FOUND)
        message(STATUS "Astyle FOUND")
        #format souce code
        **include(cmake/astyle.cmake)**        
    endif ()
else ()
    message(STATUS "Skipped using ASTYLE")
endif ()

a _astyle.cmake:_

> add_custom_target("style" COMMAND
        "${ASTYLE_EXECUTABLE}"
        --options=${CMAKE_CURRENT_SOURCE_DIR}/../astyle.options
        ${CMAKE_CURRENT_SOURCE_DIR}/../include/*.h
        ${CMAKE_CURRENT_SOURCE_DIR}/../src/*.cpp
        COMMENT "Formating source code with Astyle."
        VERBATIM
        )

**Comment by kapecp** [comment]
naco tak komplikovane:
zoberem cely astyle a dam ho do cmake/astyle.cmake a ten includnem v CMakeLists.txt
- takto bude vsetko tykajuce sa astyle na jednom mieste

**Comment by KrupsonSVK** [comment]
ok spravil som to nejak tak, je to lepsie?

**Review by marusinp — APPROVED** [review]
ok

File: CMakeLists.txt [hunk 1]
```diff
@@ -148,533 +148,27 @@ set( ALL_COMPILE_DEFINITIONS )
...
-	else ()
-		message(STATUS "3DMouse NOT FOUND")
-	endif ()
-endif ()
+#add dependencies with search module
+include(cmake/SearchDependencies.cmake)
 
 # ------------------------------------------------------------
-# -------------- DOXYGEN documentation generator--------------
+# -----Dependencies - Add custom target dependencies----------
```
Comment (kapecp) [inline CMakeLists.txt]: lepsi komentar je "Support tools - Add custom build targets"

File: cmake/astyle.cmake [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+# use "make style" to format all C++ code
+option(SKIP_ASTYLE "Skip using Astyle code formatter" OFF)
+if (NOT SKIP_ASTYLE)
+
+    find_package(Astyle)
+    if (Astyle_FOUND)
+        message(STATUS "Astyle FOUND")
+
+        # Add a custom target
+        add_custom_target("style" COMMAND
```
Comment (kapecp) [inline cmake/astyle.cmake]: treba dorobit podporu pre Windows (Win nezvladne spracovat tento prikaz, lebo je moc dlhy - obmedzenie cmd.exe) 
- vid. format.cmd -> premenovat format.cmd na run_astyle.bat
- riesit podobne ako pre CPPcheck nizsie

File: cmake/cppcheck.cmake [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+    find_package(cppcheck)
+
+    if (cppcheck_FOUND)
+        message(STATUS "Cppcheck FOUND")
+
+        file(GLOB_RECURSE ALL_SRC_FILES ${CMAKE_CURRENT_SOURCE_DIR}/../src/*.cpp)
+        if (WIN32)
+            add_custom_target(cppcheck
+                    COMMAND
+                    ${CMAKE_CURRENT_SOURCE_DIR}/../run_cppcheck.bat # NOTE: this will probably work only on Linux/OSX
```
Comment (kapecp) [inline cmake/cppcheck.cmake]: komentar netreba

File: cmake/cpplint.cmake [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+    if (CPPLINT_FOUND)
+        message(STATUS "CppLint FOUND")
+
+        FILE(GLOB_RECURSE ALL_INCLUDE_FILES ${CMAKE_CURRENT_SOURCE_DIR}/../include/*.h)
+        FILE(GLOB_RECURSE ALL_SRC_FILES ${CMAKE_CURRENT_SOURCE_DIR}/../src/*.cpp)
+
+        if (WIN32)
+            add_custom_target("cpplint"
+                    COMMAND
+                    ${CMAKE_CURRENT_SOURCE_DIR}/../cpplint.bat # NOTE: this will probably work only on Linux/OSX
```
Comment (kapecp) [inline cmake/cpplint.cmake]: - premenovat na run_cpplint.bat
- komentar netreba

**Review by kapecp — CHANGES_REQUESTED** [review]

**Review by kapecp — APPROVED** [review]

**Pull request merged**